### PR TITLE
KOGITO-2693: Embeddable editors (Milestone 1)

### PR DESCRIPTION
**Pull request by paulovmr**
**JIRA**: [KOGITO-2693](https://issues.redhat.com/browse/KOGITO-2693)

**Referenced Pull Requests**:
* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1504
* https://github.com/kiegroup/appformer/pull/1065
* https://github.com/kiegroup/kie-wb-common/pull/3464
* https://github.com/kiegroup/kogito-tooling/pull/293

**Business Central**: [WAR file](https://drive.google.com/file/d/1ty-j7N1m-Pg0Yxjoririy6jJ4ug-URX0/view?usp=sharing)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

**Comment by paulovmr** [comment]
Jenkins run fdb

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/columns/factory/dom/MonacoEditorDOMElementTest.java [hunk 1]
```diff
@@ -128,54 +122,18 @@ public void testSetupContainerComponent() {
...
+        when(domElement.uncheckedCast(any())).thenReturn(mockedElement);
 
         domElement.setupInternalComponent();
 
         verify(style).setWidth(100, PCT);
         verify(style).setHeight(100, PCT);
-        verify(editorInitializer).require(onMonacoLoaded);
-    }
-
-    @Test
```
Comment (jstastny-cz) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/columns/factory/dom/MonacoEditorDOMElementTest.java:148]: where is the equivalent of this test? Cause the functionality from `onMonacoLoaded` method now is moved (but retained) into `setupInternalComponent` method - we should not throw away test coverage.

**Review by jstastny-cz — CHANGES_REQUESTED** [review]

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/columns/factory/dom/MonacoEditorDOMElementTest.java (hunk 1 above)
Comment (paulovmr) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/columns/factory/dom/MonacoEditorDOMElementTest.java:148]: I've now moved the checks to the `setupInternalComponent` test.

**Comment by paulovmr** [comment]
Jenkins run fdb

**Comment by paulovmr** [comment]
Jenkins run fdb

**Review by karreiro — COMMENTED** [review]
Excellent work, @paulovmr. Clever approach with `appformer-js-monaco` ;-)

I've left only a minor comment on this PR. Please, let me know wdyt!

Thank you :-)

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/columns/factory/dom/MonacoEditorDOMElementTest.java (hunk 1 above)
Comment (jstastny-cz) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/columns/factory/dom/MonacoEditorDOMElementTest.java:148]: Thanks

**Review by jstastny-cz — APPROVED** [review]

**Comment by jstastny-cz** [comment]
I've run our regression suite with the BC war from FDB, no issues found.

**Review by karreiro — APPROVED** [review]
Thank you, @paulovmr.

**Review by vpellegrino — APPROVED** [review]
LGTM, good work @paulovmr

**Review by tiagobento — APPROVED** [review]

**Comment by paulovmr** [comment]
Jenkins run fdb

**Comment by paulovmr** [comment]
Jenkins run fdb

**Comment by paulovmr** [comment]
Jenkins run fdb

**Comment by paulovmr** [comment]
Jenkins run fdb

**Comment by karreiro** [comment]
Jenkins run fdb.

**Comment by karreiro** [comment]
Jenkins run fdb.

**Comment by paulovmr** [comment]
Jenkins run fdb

**Comment by paulovmr** [comment]
Jenkins run fdb

**Comment by paulovmr** [comment]
Jenkins run fdb

**Comment by paulovmr** [comment]
Jenkins run fdb

**Comment by paulovmr** [comment]
Jenkins run fdb

**Comment by paulovmr** [comment]
Jenkins run fdb

**Comment by paulovmr** [comment]
Jenkins run fdb

**Pull request merged**